apps/full-site-editing/full-site-editing-plugin/block-helpers/block-helpers.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import { getCategories } from '@wordpress/blocks';
+
+/**
+ * Accepts an array of category names and returns the first one that
+ * exists in the categories WP store. This allows for a "progressive
+ * enhancement" strategy to category names, where we just append the
+ * new category name when it's updated, and the block will still use
+ * the old ones in environments where it still exists.
+ *
+ * @param requestedCategories an array of categories.
+ * @returns {string} the first category name found.
+ * @throws {Error} if the no categories could be found.
+ */
+export function getCategoryWithFallbacks( ...requestedCategories: string[] ): string {
+	const knownCategories: Array< { slug: string } > = getCategories();
```

Comment:
Let's let the type be inferred here, we should have a [good type for `getCategories`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ab9cec4dc372cb96964c6beaf2fa48026c8751dd/types/wordpress__blocks/api/categories.d.ts#L12).
```suggestion
	const knownCategories = getCategories();
```

Reply:
:+1: Doing some type hygiene was part of my plan btw, but I started with your verbatim copy in order to work on setting up TS for Jest, as it wasn't working before.